tests/models/test_implicit_als.py

```diff
@@ -14,16 +14,18 @@
 
 import typing as tp
 
+import implicit.gpu
 import numpy as np
 import pandas as pd
 import pytest
-from implicit.als import AlternatingLeastSquares
-from implicit.cuda import HAS_CUDA
+from implicit.als import AlternatingLeastSquares as AlternatingLeastSquaresFactory
```

Comment:
Ohh, this really could lead to confusions. And I'm talking not only about the tests.

Maybe we could rename our generic type to smth like `AnyAlternatingLeastSquares` to avoid them?